modules/data.atmosphere/R/download.Drought2018.R

```diff
@@ -0,0 +1,195 @@
+#' Download ICOS Drought 2018 data
+#'
+#' Link to the product: https://www.icos-cp.eu/data-products/YVR0-4898
+#'
+#' Variables present in the output CSV file: TA_F, SW_IN_F, LW_IN_F, VPD_F, PA_F, P_F, WS_F, WD, RH, PPFD_IN, CO2_F_MDS, TS_F_MDS_1, TS_F_MDS_2, NEE_VUT_REF, LE_F_MDS, RECO_NT_VUT_REF and GPP_NT_VUT_REF
+#'
+#' @param sitename ICOS id of the site. Example - "BE-Bra"
+#' @param outfolder path to the directory where the output file is stored. If specified directory does not exists, it is created.
+#' @param start_date start date of the data request in the form YYYY-MM-DD
+#' @param end_date end date area of the data request in the form YYYY-MM-DD
+#' @param overwrite should existing files be overwritten. Default False.
+#' @return information about the output file
+#' @export
+#' @examples
+#' \dontrun{
+#' download.Drought2018("FI-Sii", "/home/carya/pecan", "2016-01-01", "2018-01-01")
+#' }
+#' @author Ayush Prasad
+#'
+download.Drought2018 <-
```

Comment:
Thanks Ayush for the PR, I'll review it closely during the week. However, I forgot about our discussion regarding ICOS data platform which could be good to remember and document here. 

In ICOS network, I think there are sites and datasets that we are interested besides Drought2018 (while the primary aim of your project is to get this dataset of course). So I was curious, how different is it to retrieve the other datasets from ICOS data platform? I guess what I'm trying to ask is, how difficult is it to design this function as a `download.ICOS` generic function where Drought2018 is one of the cases? Did you have the chance to play with a non-Drought2018 ICOS dataset through this pipeline?

Reply:
Yes, we had discussed about having a generic ICOS function that would also retrieve the drought 18 data. However, while actually implementing it I found that there aren't that many data sets that could meet our requirements. The only one that I could find was the, [ICOS  Level 2 Ecosystem Archive data](https://www.icos-cp.eu/data-products), which has all of the variables that we might need (meteorological and fluxes).  If we want to access this dataset using the function in the PR, we'd need to replace the url [here](https://github.com/PecanProject/pecan/pull/2813/files#diff-ceb68ce8213bf7a6877ac52e45eb33398b93f9521c33d6bfa74cd12779925069R77) and probably also change the output file names, no other changes would be required. 

Do you know any other relevant ICOS datasets? I can see that some sites are also publishing ecosystem data in near real-time but those are raw data and do not contain all of the variables.

In principle, it is probably possible to download all of the ICOS datasets (Ecosystem, Atmospheric and Oceanic) with a single function but then we would need the exact product id for the site and not just the site name as we are able to  do [here.](https://github.com/PecanProject/pecan/pull/2813/files#diff-ceb68ce8213bf7a6877ac52e45eb33398b93f9521c33d6bfa74cd12779925069R71-R89) And the extraction steps might also not be generalisable.

So overall I think we can either modify this `download.Drought2018` function to accommodate the Ecosystem archive data by passing the choice as a parameter or we can have something like `download.ICOS_ecosystem_archive` with a slightly higher level of abstraction. What do you think?

Reply:
Thanks Ayush, this was helpful. Yes, exactly I'm just thinking of Ecosystem datasets, particularly the Level 2 products, I think these will keep expanding besides the Drought2018 and it would be very useful to have access to those as well. 

I haven't played with their near real-time data but that also sounds very interesting, and I think as long as variable names units etc. and retrieval approaches are not too different, some of the missing variables could be dealt with.  

I agree that matching exact product ids may not be easily generalizable but your functions can hardcode the known ones and assume that it will be known in the future. In other words, I think figuring out which product id to pass your functions is easier than figuring out the rest of the pipeline, so aiming for a partially generalizable function could be valuable.

Overall, I'll have a better grasp of the situation when I review your PR, but my initial feeling would be to aim for a generic `download.ICOS` function, have L2 and Drought2018 as different cases in there as a start.